File: full_case_study_solns/_d_mass/matlab_old/hw8/massParamHW8.m
% mass - parameter file
addpath ./.. % adds the parent directory to the path
massParam % general parameters

% tuning parameters
tr = 2.2;  % tuned value of the rise time
zeta = 0.7;

% dirty derivative gain
P.sigma = 0.05;

% input constraint
P.F_max = 2;

% open loop char polynomial and poles
Delta_ol = [1,P.b/P.m,P.k/P.m];
p_ol = roots(Delta_ol);

% PD gains
wn = 2.2/tr;
Delta_cl_d = [1,2*zeta*wn,wn^2];
P.kp =  P.m*(Delta_cl_d(3)-Delta_ol(3));
P.kd = P.m*(Delta_cl_d(2)-Delta_ol(2));

fprintf('\t kp: %f\n', P.kp)
fprintf('\t kd: %f\n', P.kd)

File: full_case_study_solns/_d_mass/python_old/hw8/massParamHW8.py
# Single link arm Parameter File
import numpy as np
# import control as cnt
import sys
sys.path.append('..')  # add parent directory
import massParam as P

Ts = P.Ts  # sample rate of the controller
beta = P.beta  # dirty derivative gain
tau_max = P.F_max  # limit on control signal

#  tuning parameters
tr = 2.2
zeta = 0.707


# compute PD gains
# open loop char polynomial and poles
a1 = P.b/P.m
a0 = P.k/P.m

wn = 2.2/tr
alpha1 = 2.0*zeta*wn
alpha0 = wn**2

kp = P.m*(alpha0-a0)
kd = P.m*(alpha1-a1)

print('kp: ', kp)
print('kd: ', kd)



